cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -86,32 +168,81 @@ public void schedule(Id.Program program, SchedulableProgramType programType, Ite
 
   @Override
   public List<String> getScheduleIds(Id.Program program, SchedulableProgramType programType) {
-    return ImmutableList.of();
+    Set<String> scheduleIds;
+    synchronized (this) {
+      scheduleIds = ImmutableSet.copyOf(scheduleSubscribers.keySet());
+    }
+    ImmutableList.Builder<String> builder = ImmutableList.builder();
+    String programScheduleId = getProgramScheduleId(program, programType);
+    for (String scheduleId : scheduleIds) {
+      if (scheduleId.startsWith(programScheduleId)) {
+        builder.add(scheduleId);
+      }
+    }
+    return builder.build();
   }
 
   @Override
   public void suspendSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName) {
-
+    StreamSubscriber subscriber = scheduleSubscribers.get(getScheduleId(program, programType, scheduleName));
+    if (subscriber != null) {
+      subscriber.suspendScheduleTask(program, programType, scheduleName);
+    }
   }
 
   @Override
   public void resumeSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName) {
-
+    StreamSubscriber subscriber = scheduleSubscribers.get(getScheduleId(program, programType, scheduleName));
+    if (subscriber != null) {
+      subscriber.resumeScheduleTask(program, programType, scheduleName);
+    }
   }
 
   @Override
   public void deleteSchedule(Id.Program programId, SchedulableProgramType programType, String scheduleName) {
-
+    StreamSubscriber subscriber = scheduleSubscribers.remove(getScheduleId(programId, programType, scheduleName));
+    if (subscriber != null) {
+      subscriber.deleteSchedule(programId, programType, scheduleName);
+      synchronized (this) {
```

Comment:
I'm not sure I understand your question either. Yes a subscriber can have many tasks. One subscriber is responsible for listening to the notifications of one stream, and to poll the size of that stream. Many schedules can be attached to the same stream, hence one subscriber can have many schedule tasks. The subscriber will just pass the notifications along to the schedule tasks, which keep a state and know exactly when they should trigger the execution of a job.
There are two maps to store those objects in the StreamSizeScheduler class: ``scheduleSubscribers``, and ``streamSubscribers``. 
``streamSubscribers`` stores for each stream, its dedicated subscriber.
``scheduleSubscribers`` stores for each schedule ID, its subscriber, which is the same object as referenced by the previous ``streamSubscribers`` map. This map is needed because most APIs of the ``Scheduler`` interface only allow us to construct a scheduleId, which does not contain the stream name.
So when deleting a schedule, it is safe to always delete the entry from the ``scheduleSubscribers`` map, but we can only delete the entry from the ``streamSubscribers`` map if there is no more schedule attached to the stream. This is why there is synchronization in the ``deleteSchedule`` and ``schedule`` methods, to protect the ``streamSubscribers`` map. But this might be done in a wrong way, let me know if you see something wrong.